graphics/oidn/Portfile

```diff
@@ -0,0 +1,72 @@
+# -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4
+
+PortSystem          1.0
+PortGroup           github 1.0
+PortGroup           cmake 1.1
+
+github.setup        OpenImageDenoise oidn 1.2.0 v
+github.tarball_from releases
+extract.suffix      .src${extract.suffix}
+categories          graphics
+platforms           darwin
+license             Apache-2
+maintainers         @jasonliu-- openmaintainer
+homepage            http://www.openimagedenoise.org/
+description         denoising filters for images rendered with ray \
+                    tracing
+long_description    Open Image Denoise is a library of ${description}, \
+                    developed at Intel. Its purpose is to provide an \
+                    efficient denoising library that allows \
+                    ray-tracing-based rendering applications to \
+                    significantly reduce rendering times.
+
+checksums           rmd160  0048ae6a28a6466d81d8b62ad2382326d4e8fb6f \
+                    sha256  041f59758e79f4ea29a9b7a952f2c096426820678a5a713880b6d8a6519a75d0 \
+                    size    45522650
+
+depends_build-append \
+                    port:ispc
+depends_lib-append  port:tbb \
+                    port:openimageio
+
+compiler.cxx_standard 2011
+compiler.blacklist-append clang
+
+post-patch {
+    # Patch needed in order to compile using ISPC 1.13+
+    # (Reference: https://github.com/OpenImageDenoise/oidn/issues/60)
+    set core_math_ih ${worksrcpath}/core/math.ih
+    set status 0
+    if {[catch {system "grep -q ISPC_UINT_IS_DEFINED $core_math_ih"} results options]} {
+        set details [dict get $options -errorcode]
+        if {[lindex $details 0] eq "CHILDSTATUS"} {
+            set status [lindex $details 2]
+            if {$status == 1} {
+                reinplace "/typedef.*uint8/i\\
+#ifndef ISPC_UINT_IS_DEFINED\\
+" \
+                    $core_math_ih
+                reinplace "/typedef.*uint64/a\\
+#endif\\
+" \
+                    $core_math_ih
+            }
+        }
+    }
+}
+
+configure.args      -DOIDN_APPS_OPENIMAGEIO=ON
```

Comment:
Please append here, so as to not wipe out the general settings supplied by the cmake portgroup